docs/modules/ROOT/pages/promote-jvm-only-extensions.adoc

```diff
@@ -0,0 +1,123 @@
++
+[source,shell]
+----
+$ {
+   sed -i '/integration-test/d' "extensions-jvm/${EXT}/pom.xml"
+   sed -i "/<module>${EXT}<\/module>/d" "extensions-jvm/pom.xml"
+   git mv "extensions-jvm/${EXT}/integration-test/" "integration-tests/${EXT}"
+   git mv "extensions-jvm/${EXT}" "extensions/${EXT}"
+   sed -i -r "s/(.*)activemq(.*)/\1activemq\2\n\1${EXT}\2/g" extensions/pom.xml
+   sed -i -r "s/(.*)activemq(.*)/\1activemq\2\n\1${EXT}\2/g" integration-tests/pom.xml
+   sed -i "s/camel-quarkus-grpc-parent/camel-quarkus-integration-tests/g" "integration-tests/${EXT}/pom.xml"
+   sed -i "s/camel-quarkus-grpc-integration-test/camel-quarkus-integration-test-${EXT}/g" "integration-tests/${EXT}/pom.xml"
+   sed -i -r "s/Quarkus :: (.*) :: Integration Test/Quarkus :: Integration Test :: \1/g" "integration-tests/${EXT}/pom.xml"
+  }
+----
+
+5. Add the native profile at the end of `integration-tests/${EXT}/pom.xml`:
++
+[source,xml]
+----
+    <profiles>
+        <profile>
+            <id>native</id>
+            <activation>
+                <property>
+                    <name>native</name>
+                </property>
+            </activation>
+            <properties>
+                <quarkus.package.type>native</quarkus.package.type>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-failsafe-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>integration-test</goal>
+                                    <goal>verify</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+----
+
+6. Remove the warning link:https://quarkus.io/guides/writing-extensions#build-step-processors[build step] from `extensions/${EXT}/deployment/src/main/java/org/apache/camel/quarkus/component/${EXT}/deployment/${EXT}Processor.java`:
++
+[source,java]
+----
+    /**
+     * Remove this once this extension starts supporting the native mode.
+     */
+    @BuildStep(onlyIf = NativeBuild.class)
+    @Record(value = ExecutionTime.RUNTIME_INIT)
+    void warnJvmInNative(JvmOnlyRecorder recorder) {
+        JvmOnlyRecorder.warnJvmInNative(LOG, FEATURE); // warn at build time
+        recorder.warnJvmInNative(FEATURE); // warn at runtime
+    }
+----
+
+7. Create a native test at `integration-tests/${EXT}/src/test/java/org/apache/camel/quarkus/component/${EXT}/it/${EXT}IT.java`
+
+8. Check `extensions/${EXT}/runtime/src/main/resources/META-INF/quarkus-extension.yaml`:
+* Ensure guide is set to `"https://quarkus.io/guides/camel"`
+* Ensure keyword `camel` is present
+* Remove the `preview` status
```

Comment:
@ppalaga Would the step below not be executed, we end up with a build issue:
The step "8" is needed otherwise we have:

[ERROR] Failed to execute goal org.codehaus.gmaven:groovy-maven-plugin:2.1.1:execute (validate-extension-metadata) on project camel-quarkus: Execution validate-extension-metadata of goal org.codehaus.gmaven:groovy-maven-plugin:2.1.1:execute failed:
[ERROR] Quarkus extension metadata validation failures:
[ERROR]
[ERROR] extensions/grpc/runtime/src/main/resources/META-INF/quarkus-extension.yaml must contain a link to the guide https://quarkus.io/guides/camel
[ERROR] extensions/grpc/runtime/src/main/resources/META-INF/quarkus-extension.yaml metadata must contain a list of keywords with at least "camel" present
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

Is it a non expected build issue that would need to be fixed ?

Reply:
I was only proposing to remove the "Ensure keyword `camel` is present" line.

Reply:
Opened follow up issue #1058.